### PR TITLE
[ci] Update to OCaml 4.08.1

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

File: dev/ci/docker/bionic_coq/Dockerfile [hunk 1]
```diff
@@ -56,7 +56,7 @@ RUN opam switch create "${COMPILER}+32bit" && eval $(opam env) && \
     opam install $BASE_OPAM
 
 # EDGE switch
-ENV COMPILER_EDGE="4.08.0" \
+ENV COMPILER_EDGE="4.08.1" \
```
Comment (Zimmi48) [inline dev/ci/docker/bionic_coq/Dockerfile]: You did not change the `CACHEKEY`. Is that an overlook?

Comment (ejgallego) [reply dev/ci/docker/bionic_coq/Dockerfile]: Yes @Zimmi48 !! Sorry for that. Rebuilding both images.

**Comment by SkySkimmer** [comment]
opam doesn't find 4.08.1 on windows, I don't know when we can expect https://github.com/fdopen/opam-repository-mingw/tree/opam2/packages/ocaml-variants to be updated.

**Comment by ejgallego** [comment]
Seems updated now, will rebase.

**Comment by ejgallego** [comment]
Ups, the `c` variant has not yet been added.

**Comment by ejgallego** [comment]
Should work now.

**Comment by ejgallego** [comment]
@vbgl you may want to backport this.

**Review by vbgl — APPROVED** [review]

**Comment by vbgl** [comment]
Ping @SkySkimmer

**Review by SkySkimmer — APPROVED** [review]

**Comment by vbgl** [comment]
It seems that this PR depends on #10471; I’ve thus postponed the milestone.

**Comment by ejgallego** [comment]
> It seems that this PR depends on #10471; I’ve thus postponed the milestone.

Oh indeed, I got confused as I thought we had backported 4.08.0 dev support to 8.10, sorry about that @vbgl

**Pull request merged**